File: projecteuler/python/p050.py
from euler.mathtools import sieve_of_atkin


def main():
    limit = 10 ** 6
    primes = sieve_of_atkin(limit)

    maxima = 0
    while primes:
        sigma = 0
        count = 0
        for p in primes:
            sigma += p
            count += 1
            if sigma in primes and count > maxima:
                maxima = count
                print(sigma)
            if sigma >= limit:
                break
        primes = primes[1:]


if __name__ == '__main__':
    main()

File: projecteuler/lua/p050.lua
local mathtools = require('lua.euler.mathtools')
local helpers = require('lua.euler.helpers')

limit = 10 ^ 6
primes = mathtools.sieveOfAtkin(limit)
primeSet = helpers.set(primes)

maxima = 0
while #primes > 0 do
  sigma = 0
  count = 0
  for _, p in ipairs(primes) do
    sigma = sigma + p
    count = count + 1
    if primeSet[sigma] and count > maxima then
      maxima = count
      print(sigma)
    end
    if sigma >= limit then
      break
    end
  end
  table.remove(primes, 1)
end
